File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int n = nums.length, pointer = 0;
        for(int i = 0; i < n; i++){
            if(nums[i] != val){
                nums[pointer++] = nums[i];
            }
        }
        return pointer;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size(); 
        vector<int> ans(n);

        for(int i = 0; i < n; i++){
            ans[i] = ans[i - 1] * nums[i];
            cout << ans[i] << endl;
        }
        return ans;
    }
};
